Formula/smali.rb

```diff
@@ -1,38 +1,21 @@
 class Smali < Formula
   desc "Assembler/disassembler for Android's Java VM implementation"
   homepage "https://github.com/JesusFreke/smali"
-  url "https://bitbucket.org/JesusFreke/smali/downloads/smali-2.3.4.jar"
-  sha256 "a7c3bedc9c2a82191a30cfd1a0191f984f7b9add5a75112a7c9686d01dd99d0b"
+  url "https://github.com/JesusFreke/smali/archive/v2.3.4.tar.gz"
+  revision 1
+  sha256 "d364ebb60ac954cac7c974d72def897a373430fcd4e3349816743147fbaba375"
 
-  bottle :unneeded
-
-  resource "baksmali-jar" do
-    url "https://bitbucket.org/JesusFreke/smali/downloads/baksmali-2.3.4.jar"
-    sha256 "4896337b63a48c318de35f1f861b2a374d0f1ad6b17b6067e16b7d788e8ce4ef"
-  end
-
-  resource "baksmali" do
-    url "https://bitbucket.org/JesusFreke/smali/downloads/baksmali"
-    sha256 "5d4b79776d401f2cbdb66c7c88e23cca773b9a939520fef4bf42e2856bbbfed4"
-  end
-
-  resource "smali" do
-    url "https://bitbucket.org/JesusFreke/smali/downloads/smali"
-    sha256 "910297fbeefb4590e6bffd185726c878382a0960fb6a7f0733f045b6faf60a30"
-  end
+  depends_on "gradle" => :build
+  depends_on :java => "1.8"
 
   def install
-    resource("baksmali-jar").stage do
-      libexec.install "baksmali-#{version}.jar" => "baksmali.jar"
-    end
+    system "./gradlew", "build"
```

Comment:
This might just be a personal opinion and others may disagree, but I'm not particularly a fan of using `gradlew`, despite it being used in [a few other formulae](https://github.com/Homebrew/homebrew-core/search?q=gradlew&unscoped_q=gradlew). The whole point of `gradlew` is so that you don't need to install Gradle yet we are requiring just that with `depends_on "gradle" => :build`. So `gradlew` might just download _another_ version of Gradle and not use Homebrew's Gradle at all.

Not sure what others think about it.

Reply:
I think you're right. Have replaced it to `gradle` call.